src/api/tick-stream.js

```diff
@@ -0,0 +1,31 @@
+const DerivAPIBasic = require('@deriv/deriv-api/dist/DerivAPIBasic');
+
+const api = new DerivAPIBasic({ endpoint: 'frontend.binaryws.com', lang: 'EN', app_id: 31635 });
+const token = localStorage.getItem('token');
+
+export default async function setEnemyLevelSession() {
+    try {
+        await api.authorize(token);
+        let old_data;
+        const initial_tick = await api.ticks('R_100');
+        const floor_initial_tick = Math.floor(initial_tick.tick.ask);
+
+        sessionStorage.setItem('play_enabled', true);
+        sessionStorage.setItem('previous_tick', floor_initial_tick);
+        old_data = parseInt(sessionStorage.getItem('previous_tick'));
+
+        setInterval(async () => {
+            const new_tick = await api.ticks('R_100');
+            const floor_new_tick = Math.floor(new_tick.tick.ask);
+            const pre_calc_enemy_level = parseInt(old_data) - floor_new_tick;
```

Comment:
isn't old_data converted to integer in line 15, so why is it being converted again ?

Reply:
So, what you see in line 15 is only run once (these are the initial values), as soon as the code gets into the `setInterval` loop, it is stuck in this loop function scope. Inside the loop I have to also parseInt the sessionStorage value, because a session- or localStorage value is always a string. With every loop iteration, I read a new sessionStorage value, so I have to make sure to always parse it as an integer again.

Reply:
So, what you see in line 15 is only run once (these are the initial values), as soon as the code gets into the `setInterval` loop, it is stuck in this loop function scope. Inside the loop I have to also parseInt the sessionStorage value, because a session- or localStorage value is always a string. With every loop iteration, I read a new sessionStorage value, so I have to make sure to always parse it as an integer again.

Reply:
thank you for the clarification